File: cafeoccidente-backend/src/main/java/com/cafeoccidente/backend/purchases/future/service/FuturePurchaseService.java
package com.cafeoccidente.backend.purchases.future.service;

import com.cafeoccidente.backend.purchases.future.dto.FuturePurchaseRequest;
import com.cafeoccidente.backend.purchases.future.dto.FuturePurchaseResponse;

public interface FuturePurchaseService {

    FuturePurchaseResponse create(FuturePurchaseRequest request);

    FuturePurchaseResponse findById(Long id);
}

File: cafe-occidente-frontend/src/app/core/services/future-purchase.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';

import { API_BASE_URL } from '../config/api.config';
import { FuturePurchaseRequest, FuturePurchaseResponse } from '../models/future-purchase.model';

/** Unica responsabilidad: llamadas HTTP para "Ingresar Compras a Futuro". Sin preview ni factura -
 *  el compromiso se registra directo (ver FuturePurchaseServiceImpl, backend). */
@Injectable({ providedIn: 'root' })
export class FuturePurchaseService {
  private readonly http = inject(HttpClient);

  create(request: FuturePurchaseRequest): Observable<FuturePurchaseResponse> {
    return this.http.post<FuturePurchaseResponse>(`${API_BASE_URL}/purchases/future`, request);
  }
}
